### DISCUSSION TITLE
Installations: Growing VM knowledgebase on Discord

### DISCUSSION THREAD

**Discussion by scheals** [TheOdinProject/curriculum#27640] [General]
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### Describe your suggestion

Over the years diagnosing and solving various VM problems our learners run into, the amount of knowledge and written (and unwritten/unpinned) guides on Discord has swelled. Most of the VM issues fall into couple of categories and are easily diagnosable if one knows what to look for and the solutions are already known. 

Right now, there are some issues with VirtualBox's compatibility w/ Kernel 6.50 in Ubuntu which leads to people having performance issues. I could propose adding instructions about checking Guest Additions directly, installing them manually, changing kernels, setting a kernel default to alleviate this but the VM instructions are already the most robust we have in the lesson. So currently, there is a link to a discord pinned message that refers to all of those.

It might look like setting up a VM is the most complex installation, which it isn't - it actually is the best documented, most straightforward installation we have and the various issues that might happen during and after could easily be fixed if we had space for a knowledge base somewhere. Some of the things that could be nice to have like changing the keyboard layout (unattended installation defaults to US keyboard), changing the appearance to better fit widescreen displays and probably more.

Currently, this information lives on Discord, some written down and already linked in the lesson (like the disk enlargement instructions which would need an update to better reflect the VirtualBox 7 realities) or the aforementioned Guest Additions version checking. 

I'd be a lot easier to link to a TOP page on Discord and to a TOP page in the lesson so we have one source of truth that has to be updated. This way we also don't have to rely on one user updating the pinned post.

### Path

Foundations

### Lesson Url

https://www.theodinproject.com/lessons/foundations-installations

**Comment by scheals** [comment]
[At least one user can't access the Discord links properly](https://www.reddit.com/r/theodinproject/comments/1bay1cg/how_do_i_update_virtualbox_in_windows_to_fix/kuylo1z/) and I am experiencing the same behaviour. It seems that if one is not already a member of the Discord server, a rather mysterious view of "No text channels" comes up. Being a member of the server makes the link work as it should.

I believe this is even more of a reason to make sure Discord content is available outside of it.

**Comment by CouchofTomato** [comment]
You think this all needs to be incorporated into the lesson?

**Comment by scheals** [comment]
> You think this all needs to be incorporated into the lesson?

To the contrary, I believe that a lot of what currently is in the VM instructions should be moved out of the lesson. The "Frequent issues/questions" section in Step 4 should live somewhere else and instead of relying on Discord links for changing virtual disk size & troubleshooting GuestAdditions that content would be in that "somewhere else". I am not really sure where, though. 

Additionally, having a place like that could also offer space for even more frequent issues/questions - the ones covered in Step 4 are only part of known issues and their fixes.

**Comment by CouchofTomato** [comment]
We could have a repo on github but it would look a bit clumsy for someone just beginning to learn and might be hard to navigate.

I don't really have a good solution either at the moment. It might be worth converting this to a discussion for now since it isn't something anyone can work on it seems. What do you think?

**Comment by UniquePixels** [comment]
Perhaps leverage the Wiki functions here?

**Comment by CouchofTomato** [comment]
Yes, that is one possible option.

**Comment by scheals** [comment]
> We could have a repo on github but it would look a bit clumsy for someone just beginning to learn and might be hard to navigate.
> 
> I don't really have a good solution either at the moment. It might be worth converting this to a discussion for now since it isn't something anyone can work on it seems. What do you think?

Agree.

The only thing that comes to my mind is having a Virtualbox specific FAQ page like [the regular TOP one](https://www.theodinproject.com/faq) but this means it'd live on the app repo which sounds like mixing its concerns with curriculum repo's. 

I seem to recall that contributing to Wikis is somewhat of a different workflow so it might be a barrier for contributors as well (and might be a case of "out of sight, out of mind" generally). 

As for converting this to a discussion, sure. When I find time I'll at least try to note down what could be part of that FAQ, perhaps someone will have an easier time picking it up.

Still, some action is probably needed when it comes to accessing those Discord links - at least providing a heads up that one needs to be already part of the server for them to work would help a lot, I believe.
